build/Jenkinsfile

```diff
@@ -1,52 +1,80 @@
 #!/usr/bin/env groovy
 
-def gitCommit
-node {
-    stage('Checkout') {
-        checkout scm
-        gitCommit = sh(returnStdout: true, script: 'git rev-parse HEAD').trim()
+pipeline {
+    agent {
+        docker {
+            image 'mattermost/mattermost-build-webapp:v5.2.0'
+        }
     }
-    stage('Check Style') {
-        sh 'make check-style'
-    }
-    stage('Unit/Component Tests') {
-        sh 'make test'
-    }
-    stage('Build') {
-        sh 'make build'
-        sh 'make package'
+
+    environment {
+        GIT_COMMIT = sh(returnStdout: true, script: 'git rev-parse HEAD').trim()
     }
-    stage('Push to S3') {
-        step([$class: 'S3BucketPublisher', dontWaitForConcurrentBuildCompletion: false, entries: [[
-            bucket: 'releases.mattermost.com/mattermost-webapp/${BRANCH_NAME}',
-            excludedFile: '',
-            flatten: true,
-            gzipFiles: false,
-            keepForever: false,
-            managedArtifacts: false,
-            noUploadOnFailure: true,
-            selectedRegion: 'us-east-1',
-            showDirectlyInBrowser: false,
-            sourceFile: '*.tar.gz',
-            storageClass: 'STANDARD',
-            uploadFromSlave: false,
-            useServerSideEncryption: false,
-            userMetadata: [[key: 'Cache-Control', value: 'no-cache']]
-        ], [
-            bucket: "releases.mattermost.com/mattermost-webapp/commit/${gitCommit}",
-            excludedFile: '',
-            flatten: true,
-            gzipFiles: false,
-            keepForever: false,
-            managedArtifacts: false,
-            noUploadOnFailure: true,
-            selectedRegion: 'us-east-1',
-            showDirectlyInBrowser: false,
-            sourceFile: '*.tar.gz',
-            storageClass: 'STANDARD',
-            uploadFromSlave: false,
-            useServerSideEncryption: false,
-            userMetadata: [[key: 'Cache-Control', value: 'no-cache']]
-        ]], profileName: 'Releases', userMetadAta: []])
+
+    stages {
+        stage('Install') {
+            steps {
+                sh 'make .npminstall'
+            }
+        }
+
+        stage('Check Style') {
+            steps {
+                sh 'make check-style'
+            }
+        }
+
+        stage('Test') {
+            steps {
+                sh 'make test'
+            }
+            post {
+                always {
+                    junit 'build/test-results.xml'
+                }
```

Comment:
can you add the `cleanWs notFailBuild: true` in the post section to clean the env when the build complete? this is to avoid dirty workspaces and also to not keep data in the disk (to avoid jenkins going out of disk)